source/Mlos.Python/mlos/Optimizers/OptimizerBase.py

```diff
@@ -40,13 +40,15 @@ def get_optimizer_convergence_state(self):
     def get_surrogate_model_fit_state(self):
         return self.get_optimizer_convergence_state().surrogate_model_fit_state
 
-    @abstractmethod
-    def suggest(self, random=False, context=None) -> Point:
-        """Suggest the next set of parameters to try.
+    @trace()
+    def suggest(self, context_values_dataframe=None):
+        """ Returns the next best configuration to try.

```

Comment:
What happened to the exception and the `@abstractmethod` decorator?

Reply:
Looks like some merge issues maybe, let me double check.